File: Leetcode/Container With Most Water.cc
class Solution {
public:
    int maxArea(vector<int> &height) {
        int p = 0, q = height.size() - 1, ans = 0;
        while (p < q) {
            ans = max(ans, (q - p) * min(height[p], height[q]));
            height[p] < height[q] ? p++: q--;
        }
        return ans;
    }
};

File: Leetcode/Algorithm/python/Container With Most Water.py
class Solution:
    # @return an integer
    def maxArea(self, height):
        n = len(height)
        l, r = 0, n - 1
        ans = 0
        while l < r:
            dis = r - l 
            ans = max(ans, min(height[l], height[r]) * dis)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return ans
